### ISSUE TITLE
load of misaligned address in yh_util_import_wrapped()

### ISSUE THREAD

**Issue by invd** [Yubico/yubihsm-shell#414]
During fuzzing, I've found the following with `-fsanitize=undefined`:

> runtime error: load of misaligned address 0x7b025caee261 for type 'uint16_t' (aka 'unsigned short'), which requires 2 byte alignment

at the code position

https://github.com/Yubico/yubihsm-shell/blob/c409285e2313844adadfd0362e444530913f09a7/lib/yubihsm.c#L2797

This is undefined behavior (UB) and therefore a bug.

**Comment by invd** [comment]
@aveenismail @qpernil can you confirm this issue?

**Comment by aveenismail** [comment]
@invd Apologies for the very late response. There is now a PR containing a fix for this issue.

https://github.com/Yubico/yubihsm-shell/pull/445

**Comment by aveenismail** [comment]
I should probably explain more about https://github.com/Yubico/yubihsm-shell/pull/445:

The response data is now stored using pragma pack 1, so it will still be unaligned, but so are the responses to all other commands and elsewhere in the code. But since non of the other commands are triggering this error, using pragama pack 1 might be sufficient to solve you fuzzer issue.

Is this only a fuzzing issue or are you encountering this in your regular use?

**Comment by invd** [comment]
@aveenismail I encountered this while fuzzing on an x86_64 system, with different compilers such as `clang-16` and `clang-20` (snapshot). Based on the code, I suspect the undefined behavior would also be present and detectable on corresponding builds with `-fsanitize=undefined` during actual use of the relevant functions.

**Comment by aveenismail** [comment]
And it's triggered only in yh_util_import_wrapped() and yh_util_import_rsa_wrapped()? Not any other functions, say like import_key() or yh_util_sign_attestation_certificate() for example?

**Comment by qpernil** [comment]
https://github.com/Yubico/yubihsm-shell/commit/3dd15f3c2f75a2dfb5b1f755a4b9d37a635e28c4 is part of #444, not #445. Would you be able to test with the changes in #445 ?

**Comment by invd** [comment]
@qpernil

>https://github.com/Yubico/yubihsm-shell/commit/3dd15f3c2f75a2dfb5b1f755a4b9d37a635e28c4 is part of https://github.com/Yubico/yubihsm-shell/pull/444, not https://github.com/Yubico/yubihsm-shell/pull/445.

Yes, my point in January was that I retested with the then-newest commit version on the `master` branch and still observed the issue, not that I tested #445. 

Was the following question
> Would you be able to test with the changes in https://github.com/Yubico/yubihsm-shell/pull/445 ?

directed towards me or Aveen?
 
I'm busy with other projects and can't walk you through this low-impact issue at the moment. I recommend that you reproduce + retest the issues here internally via the mentioned combination of architecture, compiler and sanitizer. If it indeed is a false positive UB warning in `clang` as you think, it would have to be present over multiple versions.
